File: agent/app/model/mail.go
package model

type MailDomain struct {
	BaseModel
	Name       string `gorm:"not null;uniqueIndex" json:"name"`
	Status     string `gorm:"not null;default:'active'" json:"status"`
	DkimKey    string `gorm:"type:text" json:"dkimKey"`
	DnsAutoGen bool   `gorm:"default:true" json:"dnsAutoGen"`
}

type MailAccount struct {
	BaseModel
	DomainID uint   `gorm:"not null;index" json:"domainID"`
	Username string `gorm:"not null" json:"username"`
	Email    string `gorm:"not null;uniqueIndex" json:"email"`
	Quota    int    `gorm:"default:0" json:"quota"`
	Status   string `gorm:"not null;default:'active'" json:"status"`
}

type MailAlias struct {
	BaseModel
	DomainID uint   `gorm:"not null;index" json:"domainID"`
	Source   string `gorm:"not null" json:"source"`
	Target   string `gorm:"not null" json:"target"`
}

File: frontend/src/api/interface/mail.ts
import { CommonModel, ReqPage } from '.';

export namespace Mail {
    export interface Domain extends CommonModel {
        name: string;
        status: string;
        dkimKey: string;
        dnsAutoGen: boolean;
    }

    export interface DomainRes extends Domain {
        accountCount: number;
        aliasCount: number;
    }

    export interface DomainCreate {
        name: string;
        dnsAutoGen: boolean;
    }

    export interface DomainDelete {
        id: number;
        cleanupDns: boolean;
    }

    export interface Account extends CommonModel {
        domainID: number;
        username: string;
        email: string;
        quota: number;
        status: string;
    }

    export interface AccountRes extends Account {
        domainName: string;
    }

    export interface AccountSearch extends ReqPage {
        domainID?: number;
        info?: string;
    }

    export interface AccountCreate {
        domainID: number;
        username: string;
        password: string;
        quota?: number;
    }

    export interface AccountUpdate {
        id: number;
        password?: string;
        quota?: number;
    }

    export interface Alias extends CommonModel {
        domainID: number;
        source: string;
        target: string;
    }

    export interface AliasRes extends Alias {
        domainName: string;
    }

    export interface AliasSearch extends ReqPage {
        domainID?: number;
        info?: string;
    }

    export interface AliasCreate {
        domainID: number;
        source: string;
        target: string;
    }

    export interface Status {
        enabled: boolean;
        mailContainerUp: boolean;
        mailContainerName: string;
        roundcubeUp: boolean;
        roundcubeURL: string;
        version: string;
    }
}
